fineract-provider/src/main/java/org/apache/fineract/infrastructure/crypt/service/CryptographyReadPlatformService.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.crypt.service;
+
+import org.apache.fineract.infrastructure.crypt.data.CryptData;
+
+/**
+ * @author manoj
+ */
+public interface CryptographyReadPlatformService {
+    CryptData getPublicRsaKey(String type);
```

Comment:
perhaps you would like to consider the fact that this an RSA key an implementation detail, and only name this `getPublicRsaKey()` - because, in the future, there may be other types of keys you want to support?

Reply:
yes, hence the class name is Cryptography* and I am implementing only RSA now.